### ISSUE TITLE
move library.db and state.pickle to $XDG_DATA_HOME

### ISSUE THREAD

**Issue by jrobeson** [sampsyo/beets#1647]
these aren't config files and thus shouldn't be in `$XDG_CONFIG_HOME`

**Comment by sampsyo** [comment]
Good point. And if we fix this, we'll need to implement some kind of migration.

**Comment by jrobeson** [comment]
are you aware of any plugins that store non config stuff in in that directory? I see a discogs_token.json , but that seems fine

**Comment by sampsyo** [comment]
I can't think of any at the moment, but a grep for `as_filename()` should reveal configurable paths (which are currently relative to the config directory by default).

**Comment by xeruf** [comment]
Hey, this would be vital for me to properly version my configs :)

**Comment by xeruf** [comment]
depending on whether the data is transient or permanent, XDG_STATE_HOME or XDG_DATA_HOME should be considered

**Comment by emiham** [comment]
I do think the defaults should be fixed, but a nice compromise would be to add a configuration option to move state.pickle (since the library can already be configured).
